python/MyriaPythonWorker.py

```diff
@@ -0,0 +1,216 @@
+from __future__ import print_function
+import os
+import sys
+import socket
+import traceback
+import struct
+import cPickle
+import base64
+
+
+class SpecialLengths(object):
+    PYTHON_EXCEPTION_THROWN = -3
+    END_OF_STREAM = -4
+    NULL = -5
+
+
+class DataType(object):
+    INT = 1
+    LONG = 2
+    FLOAT = 3
+    DOUBLE = 4
+    BLOB = 5
+
+
+class Serializer(object):
+
+    @staticmethod
+    def read_long(stream):
+        obj = stream.read(8)
+        if not obj:
+            raise EOFError
+        return struct.unpack("!q", obj)[0]
+
+    @staticmethod
+    def read_float(stream):
+        obj = stream.read(4)
+        if not obj:
+            raise EOFError
+        return struct.unpack("!f", obj)[0]
+
+    @staticmethod
+    def read_double(stream):
+        obj = stream.read(8)
+        if not obj:
+            raise EOFError
+        return struct.unpack("!d", obj)[0]
+
+    @staticmethod
+    def read_int(stream):
+        obj = stream.read(4)
+        if not obj:
+            raise EOFError
+        return struct.unpack("!i", obj)[0]
+
+    @staticmethod
+    def write_int(value, stream):
+        stream.write(struct.pack("!i", value))
+
+    @staticmethod
+    def write_float(value, stream):
+        stream.write(struct.pack("!f", value))
+
+    @staticmethod
+    def write_double(value, stream):
+        stream.write(struct.pack("!d", value))
+
+    @staticmethod
+    def write_long(value, stream):
+        stream.write(struct.pack("!q", value))
+
+
+class PickleSerializer(Serializer):
+
+    @classmethod
+    def read_item(self, stream, itemType, length):
+        obj = None
+        if itemType == DataType.INT:
+            obj = self.read_int(stream)
+        elif itemType == DataType.LONG:
+            obj = self.read_long(stream)
+        elif itemType == DataType.FLOAT:
+            obj = self.read_float(stream)
+        elif itemType == DataType.DOUBLE:
+            obj = self.read_double(stream)
+        elif itemType == DataType.BLOB:
+            obj = self.loads(stream.read(length))
+        return obj
+
+    @classmethod
+    def read_tuple(self, stream, tuplesize):
+        datalist = []
+        for i in range(tuplesize):
```

Comment:
we will want to move to python3 sooner than later so range should be fine. Also since the tuple size isn't large enough to be felled by inefficiency of range vs xrange.